### DISCUSSION TITLE
After fill, set fields read only

### DISCUSSION THREAD

**Discussion by loust333** [gettalong/hexapdf#321] [Q&A] [answered]
Hello, 

We have a PDF which we fill out with data, and we would like afterward to set the field names as "read-only= true"

Is there a method or a way to do that. I didn't find a solution in the documentation or API. 

Best regards, 

Steve

**Comment by gettalong** [comment, marked as answer]
Making a field read only can be done by setting the corresponding flag. Some flags, like the read only flag, are available for all kinds of fields. [The documentation](https://hexapdf.gettalong.org/documentation/api/HexaPDF/Type/AcroForm/Field/index.html#class-HexaPDF::Type::AcroForm::Field-label-Field+Flags) shows the flags that are available for all fields, the class documentation for the specific field classes show the specific field flags for that kind of class.

So you would do `field.flag(:read_only)` to set it to read only.

**Reply by loust333** [reply]
Thanks :)

**Reply by gettalong** [reply]
I have added a TODO to adapt the documentation to better surface this information.
